monasca_events_api/tests/first_test.py

```diff
@@ -12,10 +12,228 @@
+            'X-TENANT-ID': '0ab1ac0a-2867-402d',
+            'X-ROLES': 'user, domainuser, domainadmin, monasca-user, monasca-agent',
+            'Accept': 'application/json',
+            'User-Agent': 'python-monascaclient',
+            'Content-Type': 'application/json'}
+
+        req.body = {}
+        req.content_type = 'application/json'
+        return req
+
+    @mock.patch('monasca_events_api.v2.events.Events._list_event')
+    @mock.patch('monasca_events_api.common.repositories.mysql.mysql_repository.mdb')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_get_pass(self, helper_tenant_id, helpers_validate, mysqlRepo, listev):
+        """GET Method success"""
+        helpers_validate.validate_authorization.return_value = True
+        listev.return_value = True
+        mysqlRepo.connect.return_value = True
+        helper_tenant_id.get_tenant_id.return_value = '0ab1ac0a-2867-402d'
+        event_id = "0ab1ac0a-2867-402d-83c7-d7087262470c"
+        eventsObj = EventsSubClass()
+        eventsObj._events_repo = EventsRepository()
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        try:
+            eventsObj.on_get(self._generate_req(), res, event_id)
+        except Exception:
+            self.fail("GET() raised HTTPInternalServerError unexpectedly")
+
+    @mock.patch('monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.events.Events._validate_event')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_post_unauthorized(self, tenantid, json, http, event, validate, kafka):
+        """POST method unauthorized  """
+        json.return_value = None
+        validate.side_effect = falcon.HTTPUnauthorized('Forbidden',
+                                                       'Tenant ID is missing a'
+                                                       'required role to '
+                                                       'access this service')
+        http.return_value = self._generate_req()
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        event.return_value = True
+
+        eventsObj = EventsSubClass()
+        eventsObj._message_queue = kafka
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        self.assertRaises(falcon.HTTPUnauthorized, eventsObj.on_post,
+                          self._generate_req(), res)
+        try:
+            eventsObj.on_post(self._generate_req(), res)
+        except Exception as e:
+            self.assertEqual(e.status, '401 Unauthorized')
```

Comment:
The assertRaises here outside of the try doesn't make any sense.  You can change that to simply "self.assertRaises(falcon.HTTPUnauthorized)" and place it into the except.